src/main/kotlin/io/github/anki/anki/repository/mongodb/CardRepository.kt

```diff
@@ -2,13 +2,64 @@ package io.github.anki.anki.repository.mongodb
 
 import io.github.anki.anki.repository.mongodb.document.MongoCard
 import org.bson.types.ObjectId
-import org.springframework.data.mongodb.repository.MongoRepository
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+import org.springframework.data.mongodb.core.query.Update
 import org.springframework.stereotype.Repository
 
 @Repository
-interface CardRepository : MongoRepository<MongoCard, ObjectId> {
+class CardRepository(
+    private val mongoTemplate: MongoTemplate,
+) {
+    private val entityClass = MongoCard::class.java
 
-    fun deleteById(id: String) {
-        deleteById(ObjectId(id))
+    fun insert(mongoCard: MongoCard): MongoCard =
+        mongoTemplate.insert(mongoCard)
+
+    fun insert(mongoCards: Iterable<MongoCard>): List<MongoCard> =
+        mongoTemplate.insertAll(mongoCards.toMutableList()).toList()
+
+    fun save(mongoCard: MongoCard): MongoCard =
+        mongoTemplate.save(mongoCard)
+
+    fun findByDeckId(deckId: ObjectId, status: MongoCard.Status = MongoCard.Status.ACTIVE): List<MongoCard> =
+        mongoTemplate.find(
+            Query(
+                Criteria.where(MongoCard.DECK_ID).`is`(deckId).and(MongoCard.STATUS).`is`(status),
+            ),
+            entityClass,
+        )
+
+    fun findById(id: ObjectId, status: MongoCard.Status = MongoCard.Status.ACTIVE): MongoCard? =
```

Comment:
I suggest to rename method to `findOnlyActiveById` and if you apply my mongo repository suggestion, then - move to base repository